client/signup/controller.js

```diff
@@ -134,45 +131,50 @@ export default {
 
 			next();
 		} else {
-			waitForHttpData( () => ( { geo: requestGeoLocation() } ) )
-				.then( ( { geo } ) => {
-					const countryCode = geo.data;
-					const localeFromParams = context.params.lang;
-					const flowName = getFlowName( context.params );
-
-					if ( flowName === 'free' && 'newOnboarding' === abtest( 'newUsersWithFreePlan' ) ) {
-						gutenbergRedirect( flowName, localeFromParams );
-						return;
-					}
-
-					if (
-						( ! user() || ! user().get() ) &&
-						-1 === context.pathname.indexOf( 'free' ) &&
-						-1 === context.pathname.indexOf( 'personal' ) &&
-						-1 === context.pathname.indexOf( 'premium' ) &&
-						-1 === context.pathname.indexOf( 'business' ) &&
-						-1 === context.pathname.indexOf( 'ecommerce' ) &&
-						-1 === context.pathname.indexOf( 'with-theme' ) &&
-						'variantUserless' === abtest( 'userlessCheckout', countryCode )
-					) {
-						removeWhiteBackground();
-						const stepName = getStepName( context.params );
-						const stepSectionName = getStepSectionName( context.params );
-						const urlWithLocale = getStepUrl(
-							'onboarding-registrationless',
-							stepName,
-							stepSectionName,
-							localeFromParams
-						);
-						window.location = urlWithLocale;
-					} else {
-						next();
-					}
-				} )
-				.catch( () => {
-					removeWhiteBackground();
-					next();
-				} );
+			next();
```

Comment:
@niranjan-uma-shankar should there be a `removeWhiteBackground();` here too?

Reply:
I haven't tested this PR yet but this looks good, `removeWhiteBackground()` is not necessary here.